tests/ntlmssptest.c

```diff
@@ -2229,6 +2229,274 @@ do { \
     return 0;
 }
 
+static size_t random_in_range(size_t min_count, size_t max_count)
+{
+    return (random() % (max_count - min_count + 1) ) + min_count;
+}
+
+static size_t generate_random_sid_str(char *dst)
+{
+    /* At least 1, 15 at the most - according to SID format */
+    size_t sub_auth_count = random_in_range(1, 15);
+    size_t offs = sprintf(dst, "S-1-5");
+    for (size_t i = 0; i < sub_auth_count; i++) {
+        offs += sprintf(dst + offs, "-%lu", (unsigned long int)random());
+    }
+    return offs;
+}
+
+static size_t generate_random_sids_list(char *dst, size_t min_count,
+                                        size_t max_count)
+{
+    size_t offs = 0;
+    size_t count = random_in_range(min_count, max_count);
+    for (size_t i = 0; i < count; i++) {
+        offs += generate_random_sid_str(dst + offs);
+        if (i < count - 1) {
+            dst[offs] = ',';
+        }
+        offs++;
+    }
+    return offs;
+}
+
+static void generate_random_binary_blob(char *dst, size_t bytes_count)
+{
+    for (size_t i = 0; i < bytes_count; i++) {
+        dst[i] = (char) (random() % 32); /* use non-printable characters */
+    }
+}
+
+static int append_sids_list_attr(const char *urn, size_t min_count,
+                                 size_t max_count, struct gssntlm_name *name,
+                                 size_t *length)
+{
+    int ret;
+    /* WBC_SID_STRING_BUFLEN is defined in wbclient.h
+     * we don't want to include here, so we use any value larger than that */
+    const size_t sid_str_alloc_size = 256;
+    char *str = malloc(max_count * sid_str_alloc_size);
+    if (!str) {
+        return ENOMEM;
+    }
+    *length = generate_random_sids_list(str, min_count, max_count);
+    ret = gssntlm_append_attr_str(urn, str, name);
+    free(str);
+    return ret;
+}
+
+static int append_binary_attr(const char *urn, size_t bytes_count,
+                              struct gssntlm_name *name)
+{
+    int ret;
+    gss_buffer_desc buf = { bytes_count, malloc(bytes_count) };
+    if (!buf.value) {
+        return ENOMEM;
+    }
+    generate_random_binary_blob(buf.value, bytes_count);
+    ret = gssntlm_append_attr(urn, &buf, name);
+    /* If append was ok, memory ownership is moved to name;
+       if failed, we need to release it here */
+    if (ret) {
+        free(buf.value);
+    }
+    return ret;
+}
+
+int test_gssapi_rfc6680(void)
+{
+    static const char *urns[] = { gssntlmssp_sids_urn, "a", "attr1", "attr2",
```

Comment:
use the string here instead of trying to reference the internal urn name, so we have a tests that prevents us from breaking the ABI inadvertently by changing the constant, this will also solve the build issues in the tests.

Reply:
yep, you are right